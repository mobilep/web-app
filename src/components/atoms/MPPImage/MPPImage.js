import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class MPPImage extends Component {

	static propTypes = {
		children: PropTypes.any,
		className: PropTypes.string,
		imageUrl: PropTypes.string,
	}

	state = {
		isImgValid: false,
	};

	checkImg (img) {

		const image = new Image();
		image.onload = () => {
			if (this._isMounted) {
				this.setState({ isImgValid: image.width > 1 });
			}
		};
		image.src = img;
		return image.width > 1;
	}

	componentDidMount () {
		this._isMounted = true;
	}

	componentWillUnmount () {
		this._isMounted = false;
	}

	componentDidUpdate ({ imageUrl }) {
		if (imageUrl !== this.props.imageUrl) {
			this.checkImg(this.props.imageUrl);
		}
	}

	UNSAFE_componentWillMount () {
		this.checkImg(this.props.imageUrl);
	}

	render () {
		const { children, className, imageUrl } = this.props;

		const style = this.state.isImgValid === false || !imageUrl
			? { backgroundColor: `rgba(243,66,60,1)` }
			: {
				background: `url(${imageUrl}) no-repeat center center / cover`,
			};

		return (
			<div
				className={className}
				style={style}>
				{children}
			</div>
		);
	}

}
