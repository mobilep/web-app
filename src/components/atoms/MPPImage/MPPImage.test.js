import React from 'react';
import { shallow } from 'enzyme';

import MPPImage from './MPPImage';

describe('Atom - MPPImage', () => {

	it('Should render', () => {
		const wrapper = shallow(<MPPImage imageUrl='' />);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render with valid image', () => {
		const wrapper = shallow(<MPPImage imageUrl='' />);
		wrapper.setState({ isImgValid: true });
		expect(wrapper).toMatchSnapshot();
	});

});

