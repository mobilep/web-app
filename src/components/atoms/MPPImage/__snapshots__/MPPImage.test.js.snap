// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - MPPImage Should render 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(243,66,60,1)",
    }
  }
/>
`;

exports[`Atom - MPPImage Should render with valid image 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(243,66,60,1)",
    }
  }
/>
`;
